lib/matplotlib/tests/test_dates.py

```diff
@@ -2,17 +2,25 @@
 import tempfile
 from unittest.mock import Mock
 
-import dateutil
+import dateutil.tz
+import dateutil.rrule
 import numpy as np
 import pytest
-import pytz
 
 from matplotlib.testing.decorators import image_comparison
 import matplotlib.pyplot as plt
 from matplotlib.cbook import MatplotlibDeprecationWarning
 import matplotlib.dates as mdates
 
 
+def __has_pytz():
+    try:
+        import pytz
+        return True
+    except ImportError:
+        return False
```

Comment:
Hm, disturbing that this line isn't being hit. Something is indirectly pulling in a `pytz` dependency in the main test suite.

Reply:
I think the problem is `pandas`. I was under the impression that the `pandas` integration tests were being run separately from the main test suite. If that's no longer the case, then this is a bit of a problem, since you could have implicit dependencies on any of the `pandas` dependencies.